app/actions/CompanyActions.js

```diff
@@ -228,27 +228,18 @@ export function editSemesterStatus(
   };
 }
 
-export function deleteSemesterStatus(
-  companyId: number,
-  semesterId: number
-): Thunk<*> {
-  return dispatch => {
-    return dispatch(
-      callAPI({
-        types: Company.DELETE_SEMESTER_STATUS,
-        endpoint: `/companies/${companyId}/semester-statuses/${semesterId}/`,
-        method: 'DELETE',
-        meta: {
-          companyId,
-          semesterId,
-          errorMessage: 'Sletting av semesterstatus feilet'
-        }
-      })
-    ).then(() => {
-      dispatch(addNotification({ message: 'Semester Status slettet' }));
-      dispatch(push(`/bdb/${companyId}/`));
-    });
-  };
+export function deleteSemesterStatus(companyId: number, semesterId: number) {
+  return callAPI({
+    types: Company.DELETE_SEMESTER_STATUS,
+    endpoint: `/companies/${companyId}/semester-statuses/${semesterId}/`,
+    method: 'DELETE',
+    meta: {
+      companyId,
+      semesterId,
+      errorMessage: 'Sletting av semesterstatus feilet',
+      successMessage: 'Semesterstatus slettet'
+    }
+  });
```

Comment:
Didn't see the reason for the push, becuse the only place you used the action was the same route as the push. It is also cleaner to have the push in the component instead of the actions, becuase it forces you to make one action for each place you are supposed to do a "thing"


And since `successMessage` is set (I set it), there will be a notification when the action succeeds.